moveit_ros/occupancy_map_monitor/include/moveit/occupancy_map_monitor/occupancy_map_monitor.h

```diff
@@ -36,127 +36,302 @@
+   */
   void setUpdateCallback(const boost::function<void()>& update_callback)
   {
     tree_->setUpdateCallback(update_callback);
   }
 
+  /**
+   * @brief      Sets the transform cache callback.
+   *
+   * @param[in]  transform_cache_callback  The transform cache callback
+   */
   void setTransformCacheCallback(const TransformCacheProvider& transform_cache_callback);
 
+  /**
+   * @brief      Set the debug flag on the updaters.
+   *
+   * @param[in]  flag  The flag
+   */
   void publishDebugInformation(bool flag);
 
+  /**
+   * @brief      Determines if active.
+   *
+   * @return     True if active, False otherwise.
+   */
   bool isActive() const
   {
     return active_;
   }
 
 private:
-  void initialize();
-
-  /** @brief Save the current octree to a binary file */
+  /**
+   * @brief      Save the current octree to a binary file
+   *
+   * @param[in]  request_header  The request header
+   * @param[in]  request         The request
+   * @param[in]  response        The response
+   *
+   * @return     True on success, False otherwise.
+   */
   bool saveMapCallback(const std::shared_ptr<rmw_request_id_t> request_header,
                        const std::shared_ptr<moveit_msgs::srv::SaveMap::Request> request,
                        std::shared_ptr<moveit_msgs::srv::SaveMap::Response> response);
 
-  /** @brief Load octree from a binary file (gets rid of current octree data) */
+  /**
+   * @brief      Load octree from a binary file (gets rid of current octree data)
+   *
+   * @param[in]  request_header  The request header
+   * @param[in]  request         The request
+   * @param[in]  response        The response
+   *
+   * @return     True on success, False otherwise.
+   */
   bool loadMapCallback(const std::shared_ptr<rmw_request_id_t> request_header,
                        const std::shared_ptr<moveit_msgs::srv::LoadMap::Request> request,
                        std::shared_ptr<moveit_msgs::srv::LoadMap::Response> response);
 
+  /**
+   * @brief      Gets the shape transform cache.
+   *
+   * @param[in]  index         The index
+   * @param[in]  target_frame  The target frame
+   * @param[in]  target_time   The target time
+   * @param[out] cache         The cache
+   *
+   * @return     True on success, False otherwise.
+   */
   bool getShapeTransformCache(std::size_t index, const std::string& target_frame, const rclcpp::Time& target_time,
                               ShapeTransformCache& cache) const;
 
-  std::shared_ptr<tf2_ros::Buffer> tf_buffer_;
-  std::string map_frame_;
-  double map_resolution_;
-  boost::mutex parameters_lock_;
-
-  OccMapTreePtr tree_;
-  OccMapTreeConstPtr tree_const_;
+  std::unique_ptr<MiddlewareHandle> middleware_handle_; /*!< The abstract interface to ros */
+  std::shared_ptr<tf2_ros::Buffer> tf_buffer_;          /*!< TF buffer */
+  Parameters parameters_;
+  std::mutex parameters_lock_; /*!< Mutex for synchronizing access to parameters */
```

Comment:
I just realized that sometimes the access to parameters is protected by mutex but most of the time it isn't. Is that a bug or am I just reading the code wrong?

Reply:
This mutex is never actually used to synchronize anything.  I think someone had a plan to do that but didn't.  I am separately working on implementing dynamic parameters for moveit2 and will clean this up in an unrelated change.